### PR TITLE
Fix LMDBReader crash due to not fully cleanup

### PR DESCRIPTION
**Pull request by bowang**
This PR fixed a bug in LMDBReader cleanup. Some pointers were not reset and caused corruption when the LMDBReader opens the next file.
A test case is constructed to verify the case.

### CODE REVIEW

**Comment by sb2nov** [comment]
Jenkins, test this please

**Comment by bowang** [comment]
The failure looks weird:
https://ci.tensorflow.org/job/tensorflow-pull-requests-gpu/7119/console
Can someone help me take a look?

File: tensorflow/python/kernel_tests/reader_ops_test.py [hunk 1]
```diff
@@ -1029,6 +1032,25 @@ def testReadFromFolder(self):
...
+    with self.test_session() as sess:
+      reader = io_ops.LMDBReader(name="test_read_from_file_repeated")
+      filename_queue = input_lib.string_input_producer([self.db_path],
+                                                       num_epochs=None)
+      key, value = reader.read(filename_queue)
+
+      coord = coordinator.Coordinator()
+      threads = queue_runner_impl.start_queue_runners(sess, coord=coord)
+      # Iterate over the lmdb 3 times.
+      for i in range(0, 3):
```
Comment (jhseu) [inline tensorflow/python/kernel_tests/reader_ops_test.py]: range(3)

File: tensorflow/python/kernel_tests/reader_ops_test.py [hunk 2]
```diff
@@ -1029,6 +1032,25 @@ def testReadFromFolder(self):
...
+        # Go over all 10 records each time.
+        for j in range(0, 10):
```
Comment (jhseu) [inline tensorflow/python/kernel_tests/reader_ops_test.py]: range(10)

**Review by jhseu — APPROVED** [review]

File: tensorflow/python/kernel_tests/reader_ops_test.py (hunk 1 above)
Comment (bowang) [reply tensorflow/python/kernel_tests/reader_ops_test.py]: Fixed. Thanks!

File: tensorflow/python/kernel_tests/reader_ops_test.py (hunk 2 above)
Comment (bowang) [reply tensorflow/python/kernel_tests/reader_ops_test.py]: Fixed. Thanks!

**Comment by bowang** [comment]
@jhseu is there anything I can do to get it merged?

**Comment by bowang** [comment]
Hi @jhseu , can we merge this PR?

**Comment by jhseu** [comment]
@caisq is on our rotation and can merge it when he has time.

**Comment by caisq** [comment]
@tensorflow-jenkins test this please

**Comment by bowang** [comment]
The CI tests seem quite weird and unrelated...

**Comment by caisq** [comment]
@tensorflow-jenkins test this please

**Comment by bowang** [comment]
Why does it fails so strangely? Can someone provide some hint?

**Comment by bowang** [comment]
I just merged the latest master. Hope the failures are resolved in master branch already. Can we kick off a jenkins test again?

**Comment by bowang** [comment]
@caisq Would you mind kicking off a CI test?

**Comment by bowang** [comment]
@jhseu can you launch a jenkins test?

**Comment by jhseu** [comment]
Jenkins, test this please

**Comment by bowang** [comment]
@jhseu Thanks for starting the test! Do you have any idea why "Ubuntu CC" is still pending?

**Comment by bowang** [comment]
I found the following error from CI:
```
ERROR: /tmpfs/src/github/tensorflow/tensorflow/contrib/image/BUILD:192:1: Couldn't build file tensorflow/contrib/image/gen_single_image_random_dot_stereograms_ops_py_wrappers_cc: 
Linking of rule '//tensorflow/contrib/image:gen_single_image_random_dot_stereograms_ops_py_wrappers_cc' failed (Exit 127): clang failed: error executing command
```
It doesn't seem to have anything to do with my changes.
Why is CI so fragile and fails for unrelated reasons?
Can anyone help me just get this five-line bug fix in?
@jhseu @caisq

**Comment by Enigma-li** [comment]
@caisq @jhseu Could anyone can help to let user get the merged branch? So that we can use LMDB as TFRecords. 

Thanks

**Pull request merged**